### DISCUSSION TITLE
bufferline: adding custom clickable buttons with callback to a specific filetype

### DISCUSSION THREAD

**Discussion by andreacfromtheapp** [LazyVim/LazyVim#2420] [Q&A] [answered]
Hi :)

bufferline allows for [custom area](https://github.com/akinsho/bufferline.nvim/blob/main/doc/bufferline.txt#L1109) elements. So far I managed to add three buttons, but they show up at the end of the editor, rather than on each buffer.

<img width="732" alt="Screenshot 2024-01-24 at 9 48 07 AM" src="https://github.com/LazyVim/LazyVim/assets/3269984/d5da773f-d1da-494a-bdbf-fd8f56c65e6a">

My goal is to add those buttons to each buffer for the `.ck` filetype and tie them to callback functions to add/replace/remove a specific buffer (by pasing the buffer id or something) to/from the list you see on the right.

How can I do that please

**Comment by dpetka2001** [comment, marked as answer]
Maybe it would be best to ask at the plugin's github repo for better insight? I myself have never done this before and am not even sure what should be done.

**Reply by andreacfromtheapp** [reply]
I realized after a while that would be better. Thanks tho, I appreciate.

**Reply by dpetka2001** [reply]
@gacallea i'd be interested in learning how to do that. So if you manage to find a solution, plz share it also here :)

**Reply by andreacfromtheapp** [reply]
ok :)

one last thing is: I asked here because bufferline doesn't have a wiki nor a discussion section. and issues/pr is not the right place.

also, this may better suit the need in question https://github.com/akinsho/bufferline.nvim/blob/main/doc/bufferline.txt#L1150

since it's the same kind of deal, I also looked how LazyVim does implement the diagnostic icons but still can't figure it out.
